lib/utils/index.js

```diff
@@ -432,22 +432,52 @@ module.exports = {
   executeOnVueComponent (context, cb) {
     const filePath = context.getFilename()
     const _this = this
+    const componentComments = []
+    const foundNodes = []
+
+    function hasNode (node) {
+      return foundNodes.some(el => el.loc.start.line === node.loc.start.line)
+    }
 
     return {
+      LineComment (node) {
+        // line comment with @vue/component
+        if (!_this.isVueComponentComment(node.value)) return
+        componentComments.push(node)
+      },
+      BlockComment (node) {
+        // block comment with @vue/component
+        if (!_this.isVueComponentComment(node.value)) return
+        componentComments.push(node)
+      },
+      ObjectExpression (node) {
+        if (!componentComments.some(el => el.loc.end.line === node.loc.start.line - 1) || hasNode(node)) return
+        foundNodes.push(node)
+        cb(node)
+      },
       'ExportDefaultDeclaration:exit' (node) {
         // export default {} in .vue || .jsx
-        if (!_this.isVueComponentFile(node, filePath)) return
+        if (!_this.isVueComponentFile(node, filePath, context) || hasNode(node)) return
```

Comment:
Why did you passed `context` to `isVueComponentFile`? And why did you added `hasNode` here? I thought current check is ok and we only needed to add comment detection before objects?